### PR TITLE
Serve sockjs.js from sockjs-client dependency, don't use default from CDN

### LINKED ISSUE #474: sockjs-client updated -> Incompatibile SockJS! Main site uses: "1.1.0", the iframe: "1.0.1"
sockjs-client has been updated two days ago to 1.1.0

When using a fresh install of webpack-dev-server, it now results in the following message in the browser: `Incompatibile SockJS! Main site uses: "1.1.0", the iframe: "1.0.1"` (I use npm 3.8.6)

Extract of dependencies:
```
`-- webpack-dev-server@1.14.1
  +-- sockjs@0.3.17
  +-- sockjs-client@1.1.0
  ...
```

sockjs@0.3.17 by default use sockjs-client@1.0.1 from a CDN. Maybe webpack-dev-server should pass the option `sockjs_url` to the SockJS server and so provide the correct version of sockjs-client without relying on a CDN ?

### PR DESCRIPTION
**Pull request by Volune**
Fixes #474 "Incompatibile SockJS" (appears when using iframe mode)

### CODE REVIEW

**Comment by marie-hughes** [comment]
@sokra, could we get some movement on this? This issue makes the hotloader unusable.

**Comment by SpaceK33z** [comment]
I don't exactly get why this should be in a separate bundle. `sockjs-client` is also included in the `client/index.bundle.js`?

**Comment by elyscape** [comment]
@SpaceK33z I'm not sure if this is still the case but, at the time this pull request and issue #474 were opened, `sockjs-client` wasn't included in `client/index.bundle.js`.

File: package.json [hunk 1]
```diff
@@ -52,7 +52,10 @@
     "ssl/"
   ],
   "scripts": {
-    "prepublish": "webpack ./client/live.js client/live.bundle.js --color --config client/webpack.config.js -p && webpack ./client/index.js client/index.bundle.js --color --config client/webpack.config.js -p",
+    "prepublish": "npm run webpack-live && npm run webpack-index && npm run webpack-sockjs",
+    "webpack-live": "webpack ./client/live.js client/live.bundle.js --color --config client/webpack.config.js -p",
```
Comment (SpaceK33z) [inline package.json]: Can you change `webpack-` to `client-` here? `webpack` doesn't tell me anything in this case.

File: package.json [hunk 2]
```diff
@@ -52,7 +52,10 @@
     "ssl/"
   ],
   "scripts": {
-    "prepublish": "webpack ./client/live.js client/live.bundle.js --color --config client/webpack.config.js -p && webpack ./client/index.js client/index.bundle.js --color --config client/webpack.config.js -p",
+    "prepublish": "npm run webpack-live && npm run webpack-index && npm run webpack-sockjs",
```
Comment (SpaceK33z) [inline package.json]: use `npm run -s` here to prevent log spam.

File: client/webpack.sockjs.config.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	output: {
+		library: "SockJS",
+		libraryTarget: "umd"
+	}
+}
```
Comment (SpaceK33z) [inline client/webpack.sockjs.config.js]: missing newline at end of file

File: client/sockjs.js [hunk 4]
```diff
@@ -0,0 +1 @@
+module.exports = require('sockjs-client');
```
Comment (SpaceK33z) [inline client/sockjs.js]: missing newline at end of file

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,4 @@
 node_modules
 /client/live.bundle.js
-/client/index.bundle.js
\ No newline at end of file
+/client/index.bundle.js
+/client/sockjs.bundle.js
```
Comment (SpaceK33z) [inline .gitignore]: missing newline at end of file

**Comment by SpaceK33z** [comment]
See comments in code above. If you update that, and fix the merge conflicts, I'll merge this in :).

**Comment by SpaceK33z** [comment]
Thanks!

**Pull request merged**